sql/expression_resolver.go

```diff
@@ -386,33 +391,69 @@ func resolveTrainClause(tc *trainClause) (*resolvedTrainClause, error) {
-		EnableCache:                      enableCache,
-		CachePath:                        cachePath,
-		Epoch:                            epoch,
-		Shard:                            shard,
-		EnableShuffle:                    enableShuffle,
-		ShuffleBufferSize:                shuffleBufferSize,
-		MaxSteps:                         maxSteps,
-		GradsToWait:                      gradsToWait,
-		TensorboardLogDir:                tensorboardLogDir,
-		CheckpointSteps:                  checkpointSteps,
-		CheckpointDir:                    checkpointDir,
-		KeepCheckpointMax:                keepCheckpointMax,
-		EvalSteps:                        evalSteps,
-		EvalStartDelay:                   evalStartDecaySecs,
-		EvalThrottle:                     evalThrottleSecs,
-		EvalCheckpointFilenameForInit:    evalCheckpointFilenameForInit,
-		PredictCheckpointFilenameForInit: predictCheckpointFilenameForInit,
-		FeatureColumns:                   fcMap,
-		ColumnSpecs:                      csMap,
-		EngineParams:                     getEngineSpec(engineParams),
-		CustomModule:                     customModel}, nil
+		IsPreMadeModel:                preMadeModel,
+		ModelName:                     modelName,
+		ModelConstructorParams:        modelParams,
+		BatchSize:                     batchSize,
+		EvalBatchSize:                 evalBatchSize,
+		DropRemainder:                 dropRemainder,
+		EnableCache:                   enableCache,
+		CachePath:                     cachePath,
+		Epoch:                         epoch,
+		Shard:                         shard,
+		EnableShuffle:                 enableShuffle,
+		ShuffleBufferSize:             shuffleBufferSize,
+		MaxSteps:                      maxSteps,
+		GradsToWait:                   gradsToWait,
+		TensorboardLogDir:             tensorboardLogDir,
+		CheckpointSteps:               checkpointSteps,
+		CheckpointDir:                 checkpointDir,
+		KeepCheckpointMax:             keepCheckpointMax,
+		EvalSteps:                     evalSteps,
+		EvalStartDelay:                evalStartDecaySecs,
+		EvalThrottle:                  evalThrottleSecs,
+		EvalCheckpointFilenameForInit: evalCheckpointFilenameForInit,
+		FeatureColumns:                fcMap,
+		ColumnSpecs:                   csMap,
+		EngineParams:                  getEngineSpec(engineParams),
+		CustomModule:                  customModel}, nil
+}
+
+func resolvePredictClause(pc *predictClause) (*resolvedPredictClause, error) {
+	modelName := pc.model
+	attrs, err := resolveAttribute(&pc.predAttrs)
+	if err != nil {
+		return nil, err
+	}
+	err = ValidateAttributes(attrs)
+	if err != nil {
+		return nil, err
+	}
+	getStringAttr := func(key string, defaultValue string) string {
+		if p, ok := attrs[key]; ok {
+			strVal, _ := p.Value.(string)
+			defer delete(attrs, p.FullName)
+			if err == nil {
+				return trimQuotes(strVal)
+			}
+			fmt.Printf("ignore invalid %s=%s, default is %v", key, p.Value, defaultValue)
```

Comment:
Remove this line or use the `log.Infof(xxx)`   : )

Reply:
Thanks. Removed.